pkg/machines/components/vmSnapshotsTab.jsx

```diff
@@ -0,0 +1,169 @@
+import { ListingTable } from "cockpit-components-table.jsx";
+import { Tooltip } from '@patternfly/react-core';
+import { InfoAltIcon } from '@patternfly/react-icons';
+
+import './vmSnapshotsTab.css';
+
+const _ = cockpit.gettext;
+
+function prettyTime(unixTime) {
+    const today = parseInt(new Date() / 1000 / 24 / 60 / 60);
+    const yesterday = today - 1;
+    const date = parseInt(unixTime / 24 / 60 / 60);
+
+    let time;
+    if (date == today) {
+        time = new Date(unixTime * 1000).toISOString()
+                .slice(11, 19);
+        return "Today " + time;
+    } else if (date == yesterday) {
+        time = new Date(unixTime * 1000).toISOString()
+                .slice(11, 19);
+        return "Yesterday " + time;
+    } else {
+        const options = { year: 'numeric', month: 'long', day: 'numeric', hour: 'numeric', minute: 'numeric', second: 'numeric' };
+        return new Date(unixTime * 1000).toLocaleDateString('en-GB', options);
+    }
+}
+
+class VmSnapshotsTab extends React.Component {
+    render() {
+        const { vm } = this.props;
+        const id = vmId(vm.name);
+
+        const emptyCaption = (<div className="no-snapshots">
+            <h3>{_("No snapshots")}</h3>
+            {_("Snapshots allow you to revert to an earlier state if something goes wrong")}
+        </div>);
+
+        let detailMap = [
+            {
+                name: _("Time"), value: (snap, snapId) => {
+                    const date = prettyTime(snap.creationTime);
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-date`}>
+                            {date}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Name"), value: (snap, snapId) => {
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-name`}>
+                            {snap.name}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("Description"), value: (snap, snapId) => {
+                    let desc = snap.description;
+                    if (!desc)
+                        desc = (<span className="snap-greyed-out">{_("No description")}</span>);
+
+                    return (
+                        <div id={`${id}-snapshot-${snapId}-description`}>
+                            {desc}
+                        </div>
+                    );
+                }
+            },
+            {
+                name: _("State"), value: (snap, snapId) => {
+                    snap.state = "shutdown";
+                    const infoTips = {
+                        shutdown: _("VM is shutting down"),
+                        "disk-snapshot": _("Disk-only snapshot"),
+                        blocked: _("Domain is blocked on resource"),
+                        crashed: _("Domain is crashed"),
+                    };
```

Comment:
@andreasn Gone the tooltip way. I added these tooltip messages for these states. WDYT? Feel free to reword them.
Other states (running, paused, shutoff) seems to me self-explanatory, thus they do not need tooltips.
![Screenshot from 2019-11-11 13-17-01](https://user-images.githubusercontent.com/42733240/68586859-e9ece780-0485-11ea-9601-00dabe2827c9.png)

Reply:
_shutoff_ needs to be _shut off_. That's the state we use on the VM overview page for a VM.